wallet-new/src/Cardano/Wallet/Server/Plugins.hs

```diff
@@ -149,34 +150,39 @@ monitoringServer :: HasConfigurations
                  -> Plugin Kernel.WalletMode
 monitoringServer (NewWalletBackendParams WalletBackendParams{..}) =
     case enableMonitoringApi of
-         True  -> pure $ \_ -> do
-             serveImpl Pos.Web.Server.application
-                       "127.0.0.1"
-                       monitoringApiPort
-                       walletTLSParams
-                       Nothing
-                       Nothing
+         True  -> pure ("monitoring worker", const worker)
          False -> []
+  where
+    worker = serveImpl Pos.Web.Server.application
+                   "127.0.0.1"
+                   monitoringApiPort
+                   walletTLSParams
+                   Nothing
+                   Nothing
 
 -- | A @Plugin@ to periodically compact & snapshot the acid-state database.
 acidStateSnapshots :: AcidState db
                    -> NewWalletBackendParams
                    -> DatabaseMode
                    -> Plugin Kernel.WalletMode
-acidStateSnapshots dbRef params dbMode = pure $ \_diffusion -> do
-    let opts = getWalletDbOptions params
-    modifyLoggerName (const "acid-state-checkpoint-plugin") $
-        createAndArchiveCheckpoints
-            dbRef
-            (walletAcidInterval opts)
-            dbMode
+acidStateSnapshots dbRef params dbMode = pure ("wallet-new acid snapshots",  const worker)
+  where
+    worker = do
+      let opts = getWalletDbOptions params
+      modifyLoggerName (const "acid-state-checkpoint-plugin") $
+          createAndArchiveCheckpoints
+              dbRef
+              (walletAcidInterval opts)
+              dbMode
 
 -- | A @Plugin@ to store updates proposal received from the blockchain
 updateWatcher :: Plugin Kernel.WalletMode
-updateWatcher = pure $ \_diffusion -> do
-    modifyLoggerName (const "update-watcher-plugin") $ do
-        w <- Kernel.getWallet
-        forever $ liftIO $ do
-            newUpdate <- WalletLayer.waitForUpdate w
-            logInfo "A new update was found!"
-            WalletLayer.addUpdate w . cpsSoftwareVersion $ newUpdate
+updateWatcher = pure ("update watcher", const worker)
+  where
+    worker = do
+      modifyLoggerName (const "update-watcher-plugin") $ do
+          w <- Kernel.getWallet
+          forever $ liftIO $ do
+              newUpdate <- WalletLayer.waitForUpdate w
+              logInfo "A new update was found!"
+              WalletLayer.addUpdate w . cpsSoftwareVersion $ newUpdate
```

Comment:
I'd rather have left this file untouched and simply add the workers' names when actually passing the "Plugin" in

https://github.com/input-output-hk/cardano-sl/blob/develop/wallet-new/src/Cardano/Wallet/Action.hs#L84-L105

(same for the Legacy ones, even though less important since they'll be removed soon enough)